### PR TITLE
Bugfix/odrib/308+310+311/fix double download+count of subscribers+switch between subscribers

### PR DESCRIPTION
**Pull request by didOstap**

### CODE REVIEW

File: client/src/components/Subscribers/index.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    changeUserFollowing: (_id: string) => void;
+    resetSubscribers: () => void;
+    followUser: (body: { _id: string }) => void;
+    unfollowUser: (body: { _id: string }) => void;
+}
+
+export class Subscribers extends React.Component<ISubscribersProps> {
+    public componentDidUpdate(prevProps: any): void {
+        if (prevProps.loggedId !== this.props.loggedId) {
+            const subscribers = this.props.path === '/profile/:username/following' ? 'following' : 'followers';
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/index.tsx]: Looks like hardcode. Could you check in another way?
Or use constants, or indexOf/find

File: client/src/components/Subscribers/index.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        }
+    }
+
+    public componentWillUnmount(): void {
+        this.props.resetSubscribers();
+    }
+
+    public getMoreFollowers = (): void => {
+        if (!this.props.allSubscribersLoaded && this.props.loggedId) {
+            const subscribers = this.props.path === '/profile/:username/following' ? 'following' : 'followers';
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/index.tsx]: Див вище

File: client/src/components/Subscribers/index.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                            </Button>
+                        </span>
+        );
+    }
+
+    public chooseSubscribers = (): JSX.Element =>
+        this.props.path === '/profile/:username/following' ?
+            <></>
+            :
+            <></>
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/index.tsx]: Нащо цей тернарник якщо вертає то саме?))

Comment (didOstap) [reply client/src/components/Subscribers/index.tsx]: забув забрати)
fixed

File: client/src/components/Subscribers/index.tsx (hunk 1 above)
Comment (didOstap) [reply client/src/components/Subscribers/index.tsx]: fixed

File: client/src/components/Subscribers/index.tsx (hunk 2 above)
Comment (didOstap) [reply client/src/components/Subscribers/index.tsx]: fixed

File: client/src/components/Subscribers/index.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    changeUserFollowing: (_id: string) => void;
+    resetSubscribers: () => void;
+    followUser: (body: { _id: string }) => void;
+    unfollowUser: (body: { _id: string }) => void;
+}
+
+export class Subscribers extends React.Component<ISubscribersProps> {
+    public componentDidUpdate(prevProps: any): void {
+        if (prevProps.loggedId !== this.props.loggedId) {
+            const subscribers = this.props.path.includes('following') ? 'following' : 'followers';
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/index.tsx]: pls move 'following' and 'followers' to const

and move to method

private getParamForSubscribers = () => this.props.path.includes('following') ? 'following' : 'followers';

Comment (didOstap) [reply client/src/components/Subscribers/index.tsx]: fixed

File: client/src/components/Subscribers/style.scss [hunk 5]
```diff
@@ -11,3 +11,12 @@
     max-width: 256px;
   }      
 }
+
+.link-style {
+  color: #808080;
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/style.scss]: variable?

File: client/src/components/Subscribers/style.scss [hunk 6]
```diff
@@ -11,3 +11,12 @@
...
+  padding: 1em;
+}
+
+.link-style:hover {
+  color: #808080;
```
Comment (InnaKutsik) [inline client/src/components/Subscribers/style.scss]: variable?

File: client/src/components/Subscribers/style.scss (hunk 5 above)
Comment (didOstap) [reply client/src/components/Subscribers/style.scss]: fixed

File: client/src/components/Subscribers/style.scss (hunk 6 above)
Comment (didOstap) [reply client/src/components/Subscribers/style.scss]: fixed

**Pull request merged**